vassal-app/src/main/java/VASSAL/counters/Embellishment.java

```diff
@@ -866,6 +866,63 @@ protected String getLocalizedCommonName(int i) {
     return translation;
   }
 
+  /**
+   * @return a list of the Decorator's string/expression fields if any (for search)
+   */
+  @Override
+  public List<String> getExpressionList() {
+    if (followProperty) {
+      return List.of(propertyName);
+    }
+    return null;
+  }
+
+  /**
+   * @return a list of any Named KeyStrokes referenced in the Decorator, if any (for search)
+   */
+  @Override
+  public List<NamedKeyStroke> getNamedKeyStrokeList() {
+    List<NamedKeyStroke> l = new ArrayList<>();
+    if (!alwaysActive) {
+      l.add(activateKeyStroke);
+    }
+    if (!followProperty) {
+      l.add(increaseKeyStroke);
+      l.add(decreaseKeyStroke);
+      l.add(resetKey);
+      l.add(rndKey);
+    }
+    return l;
+  }
+
+  /**
+   * @return a list of any Menu Text strings referenced in the Decorator, if any (for search)
+   */
+  @Override
+  public List<String> getMenuTextList() {
+    List<String> l = new ArrayList<>();
+    if (!alwaysActive) {
+      l.add(activateCommand);
+    }
+    if (!followProperty) {
+      l.add(upCommand);
+      l.add(downCommand);
+      l.add(resetCommand);
+    }
+    return l;
+  }
+
+  /**
+   * @return a list of any Message Format strings referenced in the Decorator, if any (for search)
+   */
+  @Override
+  public List<String> getFormattedStringList() {
+    if (!followProperty) {
+      return List.of(resetLevel.getFormat());
+    }
+    return null;
```

Comment:
Do you really want to return null here? You could return an empty list instead.

Reply:
Default implementation returns null anyway, and that all works, so no big deal here?